src/App/screens/Instructor/components/InstructorLessonsCallback.js

```diff
@@ -0,0 +1,43 @@
+import {Component, PropTypes} from 'react'
+import {connect} from 'react-redux'
+import {requestInstructorLessons} from '../state/actions'
+
+export default connect(
+  null,
+  {requestInstructorLessons}
+)(class extends Component {
+  
+  static propTypes = {
+    children: PropTypes.func.isRequired,
+  }
+
+  state = {
+    currentPage: 1
+  }
+
+  fetchLessons = (currentPage = 1) => {
+    const {
+      requestInstructorLessons,
+      instructor,
+      size = 10,
+      states = undefined,
+    } = this.props
+    this.setState({currentPage})
+    requestInstructorLessons({
+      lessons_url: instructor.lessons_url,
+      page: currentPage,
+      size,
+      states,
+    })
+  }
+
+  componentWillMount() {
+    this.fetchLessons()
+  }
+
+  render() {
+    const {currentPage} = this.state
+    const {children} = this.props
+    return children(currentPage, this.fetchLessons)
+  }
+})
```

Comment:
This is a "render callback" (aka Function as Child component) used to share state across multiple components.